nilearn/plotting/surface/_backend.py

```diff
@@ -268,6 +274,91 @@ def plot_surf_stat_map(
             **kwargs,
         )
 
+    def plot_img_on_surf(
+        self,
+        stat_map,
+        surf_mesh="fsaverage5",
+        mask_img=None,
+        hemispheres=None,
+        bg_on_data=False,
+        inflate=False,
+        views=None,
+        output_file=None,
+        title=None,
+        colorbar=True,
+        vmin=None,
+        vmax=None,
+        threshold=None,
+        symmetric_cbar=None,
+        cmap=DEFAULT_DIVERGING_CMAP,
+        cbar_tick_format=None,
+        **kwargs,
+    ):
+        check_params(locals())
+
+        for arg in ("figure", "axes", "engine"):
+            if arg in kwargs:
+                raise ValueError(
+                    f"plot_img_on_surf does not accept {arg} as an argument"
+                )
+
+        # TODO shouldn't we check a list for hemispheres
+        if hemispheres in (None, "both"):
+            hemispheres = ["left", "right"]
+        hemis = check_hemispheres(hemispheres)
```

Comment:
The code did not have the check if hemispheres is a list. I'm planning to put that.